app/src/main/java/com/asfoundation/wallet/backup/use_cases/UpdateWalletPurchasesCountUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.asfoundation.wallet.backup.use_cases
+
+import com.asfoundation.wallet.repository.SharedPreferencesRepository
+import com.asfoundation.wallet.wallets.usecases.GetWalletInfoUseCase
+import io.reactivex.Completable
+import io.reactivex.Single
+import javax.inject.Inject
+
+class UpdateWalletPurchasesCountUseCase @Inject constructor(
+  private val getWalletInfoUseCase: GetWalletInfoUseCase,
+  private val sharedPreferencesRepository: SharedPreferencesRepository
+) {
+
+  operator fun invoke(walletAddress: String): Completable {
+    return getWalletInfoUseCase(null, cached = false, updateFiat = false)
+      .flatMapCompletable { walletInfo ->
+        if (walletInfo.hasBackup.not()) {
+          Single.just(sharedPreferencesRepository.getWalletPurchasesCount(walletAddress))
```

Comment:
Just an idea, not very relevant in this case :) I think this type of operation for storing data should be atomic or have some sort of lock (so only one read/write at a time). Because otherwise if 2 calls to this usecase were made in quick succession, there would be a race condition to read/write to the shared prefs.